### PR TITLE
Moved all cache warming and up-to-date code into one single class

### PR DESCRIPTION
**Pull request by stefanlenselink**
And use that class in the 3 entry points (CacheWarmer, Console-job and RequestListener)

### CODE REVIEW

File: src/Component/Asset/CacheGuard.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        $this->tracker  = $tracker;
+        $this->logger   = $logger;
+    }
+
+    /**
+     * Validate the cache, check to see if it's still valid and rebuild if it's outdated.
+     */
+    public function validate()
+    {
+        if ($this->tracker->isOutdated()) {
```
Comment (iltar) [inline src/Component/Asset/CacheGuard.php]: This method should probably be named `rebuild` or something in that fashion

File: test/Bundle/CacheWarmer/WebpackCompileCacheWarmerTest.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public function testWebpackCompileCacheWarmer()
+    {
+        $guard = $this->prophesize(CacheGuard::class);
+        $guard->validate()->shouldBeCalled();
+
+        $webpack_compile_cache_warmer = new WebpackCompileCacheWarmer($guard->reveal());
+
+        //Cache warmer is optional...
+        self::assertTrue($webpack_compile_cache_warmer->isOptional());
+        $webpack_compile_cache_warmer->warmUp('/tmp/random/name');
```
Comment (haroldiedema) [inline test/Bundle/CacheWarmer/WebpackCompileCacheWarmerTest.php]: Breaks on Windows.

Use `sys_get_temp_dir` instead.

File: src/Component/Asset/CacheGuard.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
+<?php
+namespace Hostnet\Component\Webpack\Asset;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * Gards the cache, updates or creates the cache when needed.
```
Comment (haroldiedema) [inline src/Component/Asset/CacheGuard.php]: Guards*

**Comment by iltar** [comment]
Even though this is a bug fix, it does introduce a BC break

**Pull request merged**